include/downloader.h

```diff
@@ -172,7 +172,7 @@ class Downloader
   typedef std::vector<std::pair<std::string, std::string>> Options;
 
  public: // functions
-  Downloader();
+  Downloader(std::string downloadDir);
```

Comment:
1. make `explicit`
2. In addition to serving as the default download directory, it is also used as the directory where the aria2 session file is saved. In `kiwix-desktop` it is mostly (or, rather, exclusively) used in that second role, since the actual path of the download is set via `kiwix::Downloader::startDownload()` (and thus the default download directory is never used). We can make the download directory a mandatory parameter of `kiwix::Downloader::startDownload()`, then this parameter of the `Downloader` constructor will only serve as the location of the download session file.

Reply:
Sorry but I am not sure I understand. I assume you are reffering to the call being made from within `DownloadManager::startDownload` to `kiwix::Downloader::startDownload()`. Download dir is passed there inside the options parameter. Do we want to replace that or keep it and add the download dir as a flat 3rd parameter?

Reply:
I meant adding `downloadDir` as a mandatory 2nd parameter of `kiwix::Downloader::startDownload()`. Its `options` parameter will stay as an optional 3rd parameter. The download directory passed via the `downloadDir` parameter will override the download directory, if any, passed via `options`.

Reply:
So inside `kiwix::Downloader::startDownload()` will the `downloadDir` be forced inside the `options` parameter to override whatever (if any) download directory there? And from there it will be inputted in the rest of the methods as is, through `options`.

Is this correct, or am I missing the point of this?

Reply:
Yes, that's what I meant.